shared/src/test/scala/mlscript/DiffTests.scala

```diff
@@ -537,6 +539,30 @@ class DiffTests extends org.scalatest.funsuite.AnyFunSuite with org.scalatest.Pa
             }
             // generate typescript typegen block
             if (mode.generateTsDeclarations) outputSourceCode(tsTypegenCodeBuilder.toSourceCode())
+
+            if (mode.showVariance) {
+              // process typer typedefs for type defintions
+              // in current block that have at least one
+              // type variable
+              val curBlockTypeDefs = typeDefs
+                .flatMap(typeDef => ctx.tyDefs.get(typeDef.nme.name))
+                .filter(tyDef => !tyDef.tparamsargs.isEmpty)
```

Comment:
I don't think this is a good idea. You analysis also finds the variances of method type variables. It would be inconsistent not to also compute the method type variable variances of type definitions that happen not to have type parameters.

Reply:
Currently `MethodType` does not maintain a list of type variables, so a mapping is not maintained for it's variance information, even if it is calculated. So do we still want to process type defs that don't have type parameters.

Reply:
There's no reason for `MethodType` to maintain a list of type variables anyway.

> So do we still want to process type defs that don't have type parameters.

So you agree with me, then? It's not clear to me if your message is just to confirm my suggestion or make another point.

Reply:
I've already removed this check. I'm trying to understand how method type parameters variance information be used since it's not stored anywhere.

Reply:
It is, you do seem to store it in `case t: TypeVariable` where you handle _all_ type variables, not just the type def's type params.

Reply:
Aaah you're right it's storing all the `TypeVariables` it's find. So for a method we simply have to check the `tvarVariances` of the top level TypeDef.